infrastructure/kzg/src/test/java/tech/pegasys/teku/kzg/KZGTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.kzg;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public final class KZGTest {
+  private static final String MAINNET_TRUSTED_SETUP_TEST = "trusted_setups/test_mainnet.txt";
+  private static final BigInteger BLS_MODULUS =
+      new BigInteger(
+          "52435875175126190479447740508185965837690552500527637822603658699938581184513");
+  private static final int FIELD_ELEMENTS_PER_BLOB = 4096;
+  private static final int RANDOM_SEED = 5566;
+  private static final Random RND = new Random(RANDOM_SEED);
+
+  @BeforeEach
+  public void setup() {
+    KZG.resetTrustedSetup();
+  }
+
+  @Test
+  public void testKzgDoubleResetDoNotThrow() {
+    KZG.resetTrustedSetup();
+  }
+
+  @Test
+  public void testKzgLoadTrustedSetup() {
+    final String resourcePath =
+        KZGTest.class.getResource(MAINNET_TRUSTED_SETUP_TEST).toExternalForm();
+    KZG.loadTrustedSetup(resourcePath);
+    KZG.resetTrustedSetup();
+  }
+
```

Comment:
We could add another test named like `testKzgLoadTrustedSetupTwice` to verify that we cannot load a setup without resetting the previous one